app/assets/javascripts/application.js

```diff
@@ -149,6 +150,14 @@ var Search = {
     });
   },
 
+  listenForKeys: function() {
+    window.addEventListener("keyup", function(e) {
+      if ((e.key === 's' || e.key === 'S' || e.key === '/') && e.target === document.body) {
+        $('form#search input').focus();
+      }
+    }, false);
```

Comment:
Looking a bit further up in the same file, I see that jQuery's `keyup()` method is used. While this was [deprecated in jQuery 3.3](https://api.jquery.com/keyup-shorthand/), it seems that Git's home page still uses jQuery 1.7.1 (not even 1.7.2, which was [released over 12 years ago](https://blog.jquery.com/2012/03/21/jquery-1-7-2-released/), seems that there are no good processes in place to keep the Git home page up to date), where it is _not_ deprecated.

And I would suggest not installing the event on the `window`, but on the `document`; That way, event handlers of more specific elements of the document are given a chance to intercept and override the behavior.

Additionally, I would recommend testing not specifically for the `target` to be `document.body`, but merely prevent the handler's body from running when the event originated from an `<input>` element.

From what I observed in the Javascript world, I also see that `[a, b, c].includes(x)` is preferred over `x === a || x === b || x === c`.

Lastly, in Firefox, the `/` key defaults to its default search, and that cannot be overridden via a `keyup` event handler. However, it _can_ be overridden via a `keydown` handler, which I therefore think we would want here. And it is particularly important for the `/` scenario in Firefox to prevent the default action of the event.

Putting it all together, this code works for me when I run it in a Developer Console:

```js
$(document).keydown(function(e) {
  if (e.target.tagName.toUpperCase() !== 'INPUT' && ['s', 'S', '/'].includes(e.key)) {
    e.preventDefault();
    $('form#search input').focus();
  }
});
```

What would you think about using this function body instead?

Reply:
Sounds good to me! I don't think I followed the targeting problems with testing for `document.body`, but given the code's origins it's possible there's something I'm missing. Let me tweak the function body.

I'll also see about adding a "[Press s]" message to the search placeholder.

Reply:
@dscho Pushed a new version, including an update to the placeholder.